--- adodb5/session/adodb-session2.php	Sun Feb  4 21:16:12 2007
+++ /home/jstone/dev/wgm/typemonkey/libs/devblocks/libs/adodb5/session/adodb-session2.php	Sat Apr 28 21:23:17 2007
@@ -154,7 +154,7 @@
 	*/
 	/*!
 	*/
-	function driver($driver = null) 
+	static function driver($driver = null) 
 	{
 		static $_driver = 'mysql';
 		static $set = false;
@@ -174,7 +174,7 @@
 
 	/*!
 	*/
-	function host($host = null) {
+	static function host($host = null) {
 		static $_host = 'localhost';
 		static $set = false;
 
@@ -193,7 +193,7 @@
 
 	/*!
 	*/
-	function user($user = null) 
+	static function user($user = null) 
 	{
 		static $_user = 'root';
 		static $set = false;
@@ -213,7 +213,7 @@
 
 	/*!
 	*/
-	function password($password = null) 
+	static function password($password = null) 
 	{
 		static $_password = '';
 		static $set = false;
@@ -233,7 +233,7 @@
 
 	/*!
 	*/
-	function database($database = null) 
+	static function database($database = null) 
 	{
 		static $_database = '';
 		static $set = false;
@@ -252,7 +252,7 @@
 
 	/*!
 	*/
-	function persist($persist = null) 
+	static function persist($persist = null) 
 	{
 		static $_persist = true;
 
@@ -265,7 +265,7 @@
 
 	/*!
 	*/
-	function lifetime($lifetime = null) 
+	static function lifetime($lifetime = null) 
 	{
 		static $_lifetime;
 		static $set = false;
@@ -293,7 +293,7 @@
 
 	/*!
 	*/
-	function debug($debug = null) 
+	static function debug($debug = null) 
 	{
 		static $_debug = false;
 		static $set = false;
@@ -318,7 +318,7 @@
 
 	/*!
 	*/
-	function expireNotify($expire_notify = null) 
+	static function expireNotify($expire_notify = null) 
 	{
 		static $_expire_notify;
 		static $set = false;
@@ -338,7 +338,7 @@
 
 	/*!
 	*/
-	function table($table = null) 
+	static function table($table = null) 
 	{
 		static $_table = 'sessions2';
 		static $set = false;
@@ -358,7 +358,7 @@
 
 	/*!
 	*/
-	function optimize($optimize = null) 
+	static function optimize($optimize = null) 
 	{
 		static $_optimize = false;
 		static $set = false;
@@ -378,7 +378,7 @@
 
 	/*!
 	*/
-	function syncSeconds($sync_seconds = null) {
+	static function syncSeconds($sync_seconds = null) {
 		//echo ("<p>WARNING: ADODB_SESSION::syncSeconds is longer used, please remove this function for your code</p>");
 		
 		return 0;
@@ -386,7 +386,7 @@
 
 	/*!
 	*/
-	function clob($clob = null) {
+	static function clob($clob = null) {
 		static $_clob = false;
 		static $set = false;
 
@@ -405,14 +405,14 @@
 
 	/*!
 	*/
-	function dataFieldName($data_field_name = null) {
+	static function dataFieldName($data_field_name = null) {
 		//echo ("<p>WARNING: ADODB_SESSION::dataFieldName() is longer used, please remove this function for your code</p>");
 		return '';
 	}
 
 	/*!
 	*/
-	function filter($filter = null) {
+	static function filter($filter = null) {
 		static $_filter = array();
 
 		if (!is_null($filter)) {
@@ -427,7 +427,7 @@
 
 	/*!
 	*/
-	function encryptionKey($encryption_key = null) {
+	static function encryptionKey($encryption_key = null) {
 		static $_encryption_key = 'CRYPTED ADODB SESSIONS ROCK!';
 
 		if (!is_null($encryption_key)) {
@@ -443,13 +443,13 @@
 
 	/*!
 	*/
-	function _conn($conn=null) {
-		return $GLOBALS['ADODB_SESS_CONN'];
+	static function _conn($conn=null) {
+		return isset($GLOBALS['ADODB_SESS_CONN']) ? $GLOBALS['ADODB_SESS_CONN'] : null;
 	}
 
 	/*!
 	*/
-	function _crc($crc = null) {
+	static function _crc($crc = null) {
 		static $_crc = false;
 
 		if (!is_null($crc)) {
@@ -461,7 +461,7 @@
 
 	/*!
 	*/
-	function _init() {
+	static function _init() {
 		session_module_name('user');
 		session_set_save_handler(
 			array('ADODB_Session', 'open'),
@@ -476,7 +476,7 @@
 
 	/*!
 	*/
-	function _sessionKey() {
+	static function _sessionKey() {
 		// use this function to create the encryption key for crypted sessions
 		// crypt the used key, ADODB_Session::encryptionKey() as key and session_id() as salt
 		return crypt(ADODB_Session::encryptionKey(), session_id());
@@ -484,7 +484,7 @@
 
 	/*!
 	*/
-	function _dumprs($rs) {
+	static function _dumprs($rs) {
 		$conn	= ADODB_Session::_conn();
 		$debug	= ADODB_Session::debug();
 
@@ -515,7 +515,7 @@
 	// public methods
 	/////////////////////
 	
-	function config($driver, $host, $user, $password, $database=false,$options=false)
+	static function config($driver, $host, $user, $password, $database=false,$options=false)
 	{
 		ADODB_Session::driver($driver);
 		ADODB_Session::host($host);
@@ -535,7 +535,7 @@
 
 		If $conn already exists, reuse that connection
 	*/
-	function open($save_path, $session_name, $persist = null) 
+	static function open($save_path, $session_name, $persist = null) 
 	{
 		$conn = ADODB_Session::_conn();
 
@@ -590,7 +590,7 @@
 	/*!
 		Close the connection
 	*/
-	function close() 
+	static function close() 
 	{
 /*
 		$conn = ADODB_Session::_conn();
@@ -602,7 +602,7 @@
 	/*
 		Slurp in the session variables and return the serialized string
 	*/
-	function read($key) 
+	static function read($key) 
 	{
 		$conn	= ADODB_Session::_conn();
 		$filter	= ADODB_Session::filter();
@@ -655,7 +655,7 @@
 
 		If the data has not been modified since the last read(), we do not write.
 	*/
-	function write($key, $val) 
+	static function write($key, $val) 
 	{
 	global $ADODB_SESSION_READONLY;
 	
@@ -797,7 +797,7 @@
 
 	/*!
 	*/
-	function destroy($key) {
+	static function destroy($key) {
 		$conn			= ADODB_Session::_conn();
 		$table			= ADODB_Session::table();
 		$expire_notify	= ADODB_Session::expireNotify();
@@ -844,7 +844,7 @@
 
 	/*!
 	*/
-	function gc($maxlifetime) 
+	static function gc($maxlifetime) 
 	{
 		$conn			= ADODB_Session::_conn();
 		$debug			= ADODB_Session::debug();
